### PR TITLE
Task/add hip ci job

### PR DESCRIPTION
**Pull request by kab163**
# Summary

- This PR adds HIP tests to the gitlab CI

TODO:
- [x] Add a third test that uses clang instead of gcc
- [x] Fix Hip+clang test errors
- [x] Use a regex in the ctest command to exclude the tests that are known to fail

Note:

Passing gitlab CI test suite is now ***REQUIRED*** to get your PR merged.
You can initiate the gitlab CI test suite by simply commenting on your PR 
after creation with something as simple as "LGTM".

### CODE REVIEW

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by rhornung67** [comment]
LGTM

**Comment by kab163** [comment]
> LGTM

We have to get @davidbeckingsale to restart the gitlab CI (fingers crossed everything passes!)

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by rhornung67** [comment]
@kab163 the travis builds are failing out of the gate here. It looks like a specific commit of something is trying to get pulled in but it doesn't exist. Do you have time to look into this?

**Comment by kab163** [comment]
> @kab163 the travis builds are failing out of the gate here. It looks like a specific commit of something is trying to get pulled in but it doesn't exist. Do you have time to look into this?

The radiusss submodule thingy needs to be updated, but I suck at git sometimes :( and it has been tripping me up. I have a meeting with David tomorrow to resolve it. (It runs just fine if I run it locally from my laptop, but the submodule changes I made are not being pushed here correctly).

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
@adrienbernede There is an error message, "ERROR: needs python env with lxml, please set PYTHON_ENVIRONMENT_PATH" at
https://lc.llnl.gov/gitlab/radiuss/RAJA/-/jobs/198290#L1737
and it can't find some test_output.txt file - although that doesn't cause the tests to fail. Is this a problem? (I'm looking at the hip tests on corona in the CI)

**Comment by adrienbernede** [comment]
@kab163 Solved in #994, no big deal.

**Comment by kab163** [comment]
> @kab163 Solved in #994, no big deal.

Awesome, thanks!

File: scripts/gitlab/build_and_test.sh
Comment (adrienbernede) [inline scripts/gitlab/build_and_test.sh:144]: What happens if there is an unknown hip failure? I'm afraid that as-is, corona test may never exit in error in gitlab.
You can see `line 148` that I am storing results in `tests_outputs.txt` so that I can then analyze this `line 158` and possibly `exit 1`.

Comment (davidbeckingsale) [reply scripts/gitlab/build_and_test.sh:144]: Agree, the `tee` needs to go here too.

Comment (kab163) [reply scripts/gitlab/build_and_test.sh:144]: Ah, yes that makes sense. Thanks!

**Review by rhornung67 — APPROVED** [review]
Looks good to me. @davidbeckingsale should approve before merging.

**Review by davidbeckingsale — APPROVED** [review]
Looks great!

**Pull request merged**
